supervisor/updater.py

```diff
@@ -180,6 +197,28 @@ def auto_update(self, value: bool) -> None:
         """Set Supervisor auto updates enabled."""
         self._data[ATTR_AUTO_UPDATE] = value
 
+    async def _check_connectivity(
+        self, interface: str, changed: dict[str, Any], invalidated: list[str]
+    ):
+        """Check if connectivity is true and fetch data."""
+        if interface != DBUS_IFACE_NM:
+            return
+
+        connectivity_check: bool | None = changed.get(DBUS_ATTR_CONNECTION_ENABLED)
+        connectivity: bool | None = changed.get(DBUS_ATTR_CONNECTIVITY)
+
+        # If there's no connectivity checks, stop waiting for connection
+        # Else when host says we're online, attempt to fetch version data and disable listener
+        if (
+            not connectivity_check
+            or connectivity == ConnectivityState.CONNECTIVITY_FULL
```

Comment:
So I am a bit worried that this breaks in some cases where the NetworkManager connectivity check fails but somehow Supervisor is still able to connect to the internet (e.g. differences in DNS fallback logic on host side, which is used by the Host connectivity check).

That said, I think we should resolve those issues, and rely fully rely on NetworkManager's judgement.